engine/src/execution_graph/logic_controllers/BatchProcessing.cpp

```diff
@@ -123,6 +124,11 @@ kstatus TableScan::run() {
     std::vector<int> projections(schema.get_num_columns());
     std::iota(projections.begin(), projections.end(), 0);
 
+    if (provider->is_sql()) {
+      auto sql_provider = std::dynamic_pointer_cast<ral::io::abstractsql_data_provider>(provider);
+      sql_provider->set_column_indices(projections);
```

Comment:
this logic of setting the column indices that you want should be handled by the CacheDataIO. all the other data types handle it there.

Reply:
@felipeblazing what do you think?

Reply:
Done, the logic is now in the kernel ctors and the kernel::run is agnostic to the data provider cc @felipeblazing @gcca